src/cpp/include/openvino/genai/vision_encoder.hpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/processor_config.hpp"
+#include <openvino/openvino.hpp>
+
+namespace ov::genai {
+/// @brief A pair describing image size.
+struct HeightWidth {
+    /// @brief Height of a corresponding image.
+    size_t height;
+    /// @brief Width of a corresponding image.
+    size_t width;
+};
+
+/// @brief Embeddings of a given image. The number of slices is no
+/// greater than ProcessorConfig's max_slice_nums.
+struct EncodedImage {
+    /// @brief Embeddings of a resized image based on ProcessorConfig's
+    /// scale_resolution. The tensor's shape is
+    /// [N, H*W, hidden_size]. [N, 1014, 1152] is a possible example for
+    /// openbmb/MiniCPM-V-2. Only batch 1 is supported.
+    ov::Tensor resized_source;
+    /// @brief A size of an image used to compute embeddings for
+    /// divided by ProcessorConfig's patch_size.
+    HeightWidth resized_source_size;
```

Comment:
we can extract these values from `resized_source.get_shape()[2-3]`

Reply:
`resized_source` layout is `[N, H*W, hidden_size]`. Separate height and width info is lost.